### PR TITLE
[Bridge][Twig] Update getSession method in AppVariable.php

### PR DESCRIPTION
**Pull request by Arman-Hosseini**
| Q             | A
| ------------- | ---
| Branch?       | 3.4 
| Bug fix?      | yes
| New feature?  | no 
| BC breaks?    | no     
| Deprecations? | no 
| Tests pass?   | yes    
| Fixed tickets | N/A   
| License       | MIT
| Doc PR        | N/A

### CODE REVIEW

File: src/Symfony/Bridge/Twig/AppVariable.php [hunk 1]
```diff
@@ -109,13 +109,11 @@ public function getRequest()
      */
     public function getSession()
     {
-        if (null === $this->requestStack) {
+        if ($request = $this->getRequest() && !$request->hasSession()) {
```
Comment (yceruto) [inline src/Symfony/Bridge/Twig/AppVariable.php]: missing parentheses around `$request = $this->getRequest()`

**Comment by nicolas-grekas** [comment]
Why this change?

**Comment by Arman-Hosseini** [comment]
Hello @nicolas-grekas .
Here, the "session" was not checked and only the existence of the "request" object was checked.

**Comment by yceruto** [comment]
In fact, this is a BC break, we can't throw an exception now when we didn't before. Also it's fine returning `null` when you call to `app.session`. :-1:

**Comment by stof** [comment]
The only case throwing an exception before was when not injecting the request_stack.

**Comment by fabpot** [comment]
Let's close then.

**Pull request closed without merging**